### PR TITLE
afio: init at 2.5.1

### PR DESCRIPTION
**Pull request by phi-gamma**
Signed-off-by: Philipp Gesang <phg@phi-gamma.net>

###### Motivation for this change

Add useful package.

###### Things done

- [x ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/archivers/afio/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl } :
+
+stdenv.mkDerivation rec {
+  version = "2.5.1";
+  name = "afio";
```
Comment (joachifm) [inline pkgs/tools/archivers/afio/default.nix]: The version needs to be a part of `name`, as in `name = "afio-${version}"`; the attribute itself currently holds no special meaning.

File: pkgs/tools/archivers/afio/afio-2.5.1-install.patch [hunk 2]
```diff
@@ -0,0 +1,33 @@
+--- p1/Makefile.orig	2017-02-14 21:40:20.404249126 +0100
```
Comment (joachifm) [inline pkgs/tools/archivers/afio/afio-2.5.1-install.patch]: Is this your patch (i.e., no external source)?

**Review by joachifm — CHANGES_REQUESTED** [review]

**Comment by phi-gamma** [comment]
2017-02-18 (Saturday), notifications@github.com (Joachim F.):

> joachifm requested changes on this pull request.

Thanks for the feedback.

> > @@ -0,0 +1,27 @@
> +{ stdenv, fetchurl } :
> +
> +stdenv.mkDerivation rec {
> +  version = "2.5.1";
> +  name = "afio";
> 
> The version needs to be a part of `name`, as in `name = "afio-${version}"`; the attribute itself currently holds no special meaning.

Fixed, see the revised version.

> > @@ -0,0 +1,33 @@
> +--- p1/Makefile.orig	2017-02-14 21:40:20.404249126 +0100
> 
> Is this your patch (i.e., no external source)?

It is.

**Comment by fpletz** [comment]
> Fixed, see the revised version.

What revision version? Did you forget to push?

**Comment by phi-gamma** [comment]
2017-02-21 (Tuesday), notifications@github.com (Franz Pletz):

> > Fixed, see the revised version.
> 
> What revision version? Did you forget to push?

Good question. I pushed after rebasing onto the then current
nixpkgs master. The commit is there:

    https://github.com/phi-gamma/nixpkgs/commit/b4e9bcfc12cba74f8fb4804c170658db634e5d0a

I just killed the branch and replaced it with my local head so
please check again.

**Review by joachifm — APPROVED** [review]

**Comment by phi-gamma** [comment]
2017-02-22 (Wednesday), notifications@github.com (Joachim F.):

> Merged #22812.

Great, thanks!

**Pull request merged**